frontend/appflowy_flutter/lib/plugins/database/application/cell/bloc/relation_row_search_bloc.dart

```diff
@@ -45,12 +58,56 @@ class RelationRowSearchBloc
 
   void _updateFilter(String filter, Emitter<RelationRowSearchState> emit) {
     final rows = [...allRows];
+
     if (filter.isNotEmpty) {
       rows.retainWhere(
-        (row) => row.name.toLowerCase().contains(filter.toLowerCase()),
+        (row) =>
+            row.name.toLowerCase().contains(filter.toLowerCase()) ||
+            row.name.isEmpty &&
+                LocaleKeys.grid_row_titlePlaceholder
+                    .tr()
+                    .toLowerCase()
+                    .contains(filter.toLowerCase()),
       );
     }
-    emit(state.copyWith(filter: filter, filteredRows: rows));
+
+    final focusedRowId = rows.isEmpty
+        ? null
+        : rows.any((row) => row.rowId == state.focusedRowId)
+            ? state.focusedRowId
+            : rows.first.rowId;
+
+    emit(
+      state.copyWith(
+        filteredRows: rows,
+        focusedRowId: focusedRowId,
+      ),
+    );
+  }
+
+  void _focusOption(bool previous, Emitter<RelationRowSearchState> emit) {
+    if (state.filteredRows.isEmpty) {
+      return;
+    }
+
+    final rowIds = state.filteredRows.map((e) => e.rowId).toList();
+
+    if (state.focusedRowId == null) {
+      emit(
+        state.copyWith(
+          focusedRowId: previous ? rowIds.last : rowIds.first,
+        ),
+      );
+      return;
+    }
+
+    final currentIndex = rowIds.indexWhere((id) => id == state.focusedRowId);
+
+    final newIndex = currentIndex == -1
+        ? 0
+        : (currentIndex + (previous ? -1 : 1)) % rowIds.length;
+
+    emit(state.copyWith(focusedRowId: rowIds[newIndex]));
```

Comment:
```suggestion
  void _focusOption(bool previous, Emitter<RelationRowSearchState> emit) {
    if (state.filteredRows.isEmpty) {
      return;
    }

    final rowIds = state.filteredRows.map((e) => e.rowId).toList();
    final currentIndex = state.focusedRowId == null
        ? -1
        : rowIds.indexWhere((id) => id == state.focusedRowId);

    // If the current index is -1, it means that the focused row is not in the list of row ids.
    // In this case, we set the new index to the last index if previous is true, otherwise to 0.
    final newIndex = currentIndex == -1
        ? (previous ? rowIds.length - 1 : 0)
        : (currentIndex + (previous ? -1 : 1)) % rowIds.length;

    emit(state.copyWith(focusedRowId: rowIds[newIndex]));
```